### PR TITLE
[Gekidou] upgrade to RN 68 (old arch)

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Upgrades to RN 68 and other dependencies.

To run on Android we will need to upgrade jdk to use version 11, with homebew on macOS you could just install `openjdk@11`.

For my specific case I have in my profile
```bash
export JAVA_8_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_77.jdk/Contents/Home
export JAVA_11_HOME=/usr/local/opt/openjdk@11

alias java8='export JAVA_HOME=$JAVA_8_HOME'
alias java11='export JAVA_HOME=$JAVA_11_HOME'
```

That way I can switch between versions of the jdk super quick.

Once we decide to move the `gekidou` branch as the main branch, we definitely need to update docs.

```release-note
NONE
```

### CODE REVIEW

**Review by avinashlng1080 — APPROVED** [review]
LGTM.   Approving it so as not to block this PR.  I'll be running them on iOS and Android and in case there is any issue, I will list them here.

File: test/setup.ts [hunk 1]
```diff
@@ -173,14 +174,54 @@ jest.mock('react-native-vector-icons', () => {
...
+            originalFilepath: path,
+            isFile: () => true,
+            isDirectory: () => false,
+        }),
+    };
+
+    return RNFS;
+});
+
+// jest.mock('expo-file-system', () => ({
```
Comment (larkox) [inline test/setup.ts:217]: Any need to keep this comment?

File: ios/Mattermost/main.m [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 #import "AppDelegate.h"
 
-int main(int argc, char * argv[]) {
+int main(int argc, char* argv[])
```
Comment (larkox) [inline ios/Mattermost/main.m:5]: Is this change intended?

**Review by larkox — APPROVED** [review]
LGTM, just two small things.

File: ios/Mattermost/main.m (hunk 2 above)
Comment (enahum) [reply ios/Mattermost/main.m:5]: yes, taken from the RN template

Comment (enahum) [reply ios/Mattermost/main.m:5]: see the upgrade helper for more details https://react-native-community.github.io/upgrade-helper/?from=0.67.4&to=0.68.0

File: test/setup.ts (hunk 1 above)
Comment (enahum) [reply test/setup.ts:217]: will remove

**Review by josephbaylon — CHANGES_REQUESTED** [review]
Just so it is visible here, there's currently a crash issue with push notifications with android.

**Review by josephbaylon — APPROVED** [review]
LGTM. Smoke tested both iOS and Android. The notifications issue found is fixed. Also tested building iOS and Android (java 11).
